### ISSUE TITLE
[Question] Xamarin.Forms end of life?

### ISSUE THREAD

**Issue by Tommigun1980** [xamarin/Xamarin.Forms#14205]
Hi.

I can't help but notice that there hasn't really been any commits to Xamarin.Forms for the last few months.
I understand that .NET MAUI is being worked on but let's be honest, it won't release for a good while and once it releases it will take at least a year before existing applications can even think about migrating to it. So MAUI is at least two years away for us developers.

Will Xamarin.Forms be updated during these years, or should Xamarin.Forms be deemed end-of-life as of now?

[https://github.com/xamarin/Xamarin.Forms/wiki/Feature-Roadmap](https://github.com/xamarin/Xamarin.Forms/wiki/Feature-Roadmap) says Xamarin.Forms 5 will be supported until November 2022. Is this still the case, or has Xamarin.Forms been abandoned prematurely in favour of .NET MAUI? If so then a realistic scenario is that any live applications, or applications currently in development for Xamarin.Forms, will need to be rewritten with a completely different SDK as developers can't just close shop for a few years. These applications will almost certainly not return to MAUI.

If nothing else, could the existing pull requests be merged? There's a ton of fixes there that people have been waiting for for a long time. Just take a week off from developing MAUI to merge them, or please update the roadmap to reflect that development has fully shifted to MAUI, and that the 2 year support commitment no longer holds.

Thank you so much for informing us on the roadmap and Xamarin.Forms support commitment!

**Comment by Tommigun1980** [comment]
If nothing else, could the existing pull requests be merged? There's a ton of fixes there that people have been waiting for for a long time. Just take a week off from developing MAUI to merge them, or please update the roadmap to reflect that development has fully shifted to MAUI.

**Comment by giuseppenovielli** [comment]
cc @davidortinau Please some Feedback.
We love .NET MAUI, but please spend some time also at Xamarin.Forms Bug fixing until .NET MAUI and all libraries become stable for .NET MAUI!

**Comment by giuseppenovielli** [comment]
#13997

**Comment by giuseppenovielli** [comment]
#13997

**Comment by Tommigun1980** [comment]
I am a bit disappointed in the fact that the 2-year support commitment doesn't seem to hold.
It is not something that can just be glossed over as the support commitment is very much a deciding factor when selecting a technology. If developers are hung out to dry it is certain that they won't return for any other initiatives from Microsoft.

MAUI may be the future of Xamarin.Forms but it is very much a **distant** future for us developers. It will take years before the technology is mature enough to use, before all libraries we are dependent on have been ported (if ever), and it just doesn't make sense for a lot of us completely revamp our applications.

This is the reason that the support commitment must hold and it is unprecedented for a large corporation not to uphold it.
I beg you guys to keep on supporting Xamarin.Forms as it will kill a lot of businesses if you don't.

Thank you again, and eagerly awaiting an update on this.

**Comment by Tommigun1980** [comment]
I am a bit disappointed in the fact that the 2-year support commitment doesn't seem to hold.
It is not something that can just be glossed over as the support commitment is very much a deciding factor when selecting a technology. If developers are hung out to dry it is certain that they won't return for any other initiatives from Microsoft.

MAUI may be the future of Xamarin.Forms but it is very much a *distant* future for us developers. It will take years before the technology is mature enough to use, before all libraries we are dependent on have been ported (if ever), and it just doesn't make sense for a lot of us completely revamp our applications. We don't pick a technology because it's cool, what we need is longevity, so we can keep on pursuing whatever our business goals are.

This is the reason that the support commitment must hold and it is unprecedented for a large corporation not to uphold it.
I beg you guys to keep on supporting Xamarin.Forms as it will kill a lot of businesses if you don't.

Thank you again, and eagerly awaiting an update on this.

**Comment by davidaramantSEP** [comment]
I have to agree; I am very unhappy with the support situation.  My team is stuck on Forms 4.8 because of _multiple_ crashing issues with XF5 that we cannot work around. It's not like the version we are on is a panacea either; `CollectionView` in particular is a mess (a control which we used because Microsoft said it was The New Way). I am really scared we are going to run into something where we **need** something from a later package version because we would be totally hosed.

**Comment by davidortinau** [comment]
> Will Xamarin.Forms be updated during these years, or should Xamarin.Forms be deemed end-of-life as of now?

Xamarin.Forms will receive critical updates through November 2022, a year after its successor .NET MAUI releases.

https://dotnet.microsoft.com/platform/support/policy/xamarin

> If nothing else, could the existing pull requests be merged?

Which PRs are a priority for you? I'll discuss with the team and see what we can do to get another service release out soon.

> It will take years before the technology is mature enough to use, before all libraries we are dependent on have been ported (if ever), 

.NET MAUI is Xamarin.Forms vNext in this context. What time frame for transitioning would be comfortable for you? What would you expect to happen in Xamarin.Forms when iOS 15, 16, 17 ships?

**Comment by developer9969** [comment]
@davidortinau  hi David is good that you reply ,XF at the moment is every unstable especially the collection view!!!
We updated to xf5 but we had no end problems with the collection view with resizing,grouping etc... in quite few places we had to replace it with bindable layout. Every upgrade is a worry of what might be broken... you need a bigger team Microsoft has to invest more in mobile.please   sort the collection view as after the grid is the most important control.

**Comment by Tommigun1980** [comment]
Thank you for the answer @davidortinau.

> Which PRs are a priority for you? I'll discuss with the team and see what we can do to get another service release out soon.

If I compile a list, will it actually be worth my time, and where do you want the list to be sent? I don't want to clutter this ticket.

On a high level there's a lot of pull requests that haven't been merged that fix bugs with the SwipeView and fix memory leaks.
The CollectionView requires a lot of fixes for which there are no pull requests yet. The component is not usable and we can not live two years without fixes to it.

> .NET MAUI is Xamarin.Forms vNext in this context. What time frame for transitioning would be comfortable for you?

It will take over half a year before it's released, and then at least a year or two before it's stable enough to use in production. Libraries will also take a year or two before they are updated, so I don't think it's possible to transition to it at least for two years, one year at an absolute minimum.

> What would you expect to happen in Xamarin.Forms when iOS 15, 16, 17 ships?

I would absolutely expect Xamarin.Forms be updated to support whatever OS releases are shipped during its support period. So until November 2022 Xamarin.Forms has to support any new iOS and Android versions. This is how any technology I have ever used has been supported and are the epitome or "critical fixes". If you can't release your app on the various app stores the technology is not supported.

Thank you.

**Comment by davidaramantSEP** [comment]
I too would expect that Xamarin Forms works with iOS 15. Obviously I would not expect any brand new cross-platform APIs, but Xamarin.iOS and Xamarin.Android better support new versions.

By saying that Xamarin Forms 5 was the last major release with no new significant functionality, you've set the expectation that this is supposed to be a really solid release. That has certainly not been my impression of XF5.

I really _hope_ Maui will be a straight-forward upgrade but I'm not counting on that. The reality is that the upgrade _will_ take money & time. Even if it's a perfect framework from day one, some companies will have to defer the upgrade for budgetary reasons.

**Comment by Depechie** [comment]
@davidortinau in ref to what PR's we are waiting for: https://github.com/xamarin/Xamarin.Forms/pull/13166 ( been ready for review since Januari )

**Comment by BlueRaja** [comment]
I mean, there's **two hundred** open PRs, most of which are from the past year but have not even been reviewed yet.

So a good start would be to have someone go through those and merge/close/request changes.  There must be a ton of low-hanging fruit there.

**Comment by inimirpaz** [comment]
Not to mention the 28+ days worth of "unverified" issues opened lately. What kind of support can we expect for XF 5 till Nov 2022? Major crash issues only or minor (eg layout/rendering issues) as well?

**Comment by davidortinau** [comment]
@Tommigun1980 you can email me david.ortinau@microsoft.com. I will review them with the team and prioritize those that meet the bar for a service release (regressions, critical blockers, high impact, etc.).

Xamarin.Android, Xamarin.iOS, and Xamarin.Forms will bump to new versions of Android and iOS until vNext is released, which in this case is .NET 6 in November 2021. From that point, new OS bumps will be via .NET 6+. This is a continuation of the same policy we've had for years: https://dotnet.microsoft.com/platform/support/policy/xamarin

@inimirpaz critical issues is the bar for what we are addressing right now.

Issues below that bar will be prioritized for fixing in .NET MAUI. 

Any PRs open that don't make it into Xamarin.Forms will be asked to move and retarget to .NET MAUI.

We are all in an important transition to being a unified part of .NET that will pay off in the near future. Please continue to raise specific issues to me and I'll do my best to prioritize them and guide resolution while we have a narrow focus on what we do in the Xamarin.Forms repository right now.

**Comment by davidaramantSEP** [comment]
What defines "critical? "

**Comment by developer9969** [comment]
There are 269 open Collection View issues and some of them are critical to us, Grouping- Resizing- Spacing- Empty view 
you would agree that the collection view is an extremely important component and I am sure many people would appreciate if you could go through those and fix some critical ones,

Also @Tommigun1980  did raise quite a few collectionView issues at the time , and that allowed us to avoid the control 
Shall we revert to ListView?

**Comment by AlleSchonWeg** [comment]
If the Xamarin/Maui Team fixes issues here, these issues will not appear in maui. So win win.

**Comment by MagicAndre1981** [comment]
> What defines "critical? "

this is the point. According to 80/20 rule, 80% of devs use only 20% of the XF framework, so all have different critical issues which hurt them. Microsoft doesn't use XF for their inbox Windows and cross platform apps like outlook, mobile office, so they don't know what developers need.

**Comment by Huaba93** [comment]
@MagicAndre1981 I agree with you 100%. Microsoft needs to use Xamarin/MAUI for their own products. That's what Google is doing with Flutter. That's the only way to build a great Framework.

**Comment by MagicAndre1981** [comment]
> @MagicAndre1981 I agree with you 100%. Microsoft needs to use Xamarin/MAUI for their own products.

yes, this is the point. Here Microsoft would detect the same issues like we during coding and would fix them much faster

**Comment by jsuarezruiz** [comment]
We are reviewing and merging PRs preparing the next Service Release. Will share detailed info (date, etc) as soon as possible.

**Comment by AndreasReitberger** [comment]
> We are reviewing and merging PRs preparing the next Service Release. Will share detailed info (date, etc) as soon as possible.

Hi @jsuarezruiz @PureWeen @davidortinau,
thanks for continuing working on Xamarin.Forms. I know it's a lot of work, however can you estimate a date when the SR4 is ready? According to the project here (https://github.com/xamarin/Xamarin.Forms/projects/86) it's about 43 issues still to be fixed.
As I can see, @PureWeen already fixed the AndroidX dependency and iOS TitleView issue. Those issues are vital to be fixed. Maybe you can consider to split the project and release SR4 sooner, which is fixing the vital issues, and put the rest into SR5?

Thanks for your great work and contribution!

**Comment by KazysNoobiys** [comment]
It is very sad to see the current state of the XF. Our team switched to XF 5 and encountered performance issues with CollectionView. 
At this point, CollectionView is absolutely not ready for real-world use. This has forced us to go back to ListView. But the bad news didn't end there! 
When switching to ListView, we ran into regression. In some scenarios, in debug mode, ListView absolutely nightmarishly slowed down. The corresponding issue is already open. https://github.com/xamarin/Xamarin.Forms/issues/14218 . 

Translated with www.DeepL.com/Translator (free version)

**Comment by Tommigun1980** [comment]
Pinging about this again. Microsoft has promised to support Xamarin.Forms but unfortunately I am not seeing it at the moment.

There’s a lot of developers in a really bad position right now.

Please open a dialogue with us.

**Comment by MagicAndre1981** [comment]
they released a broken SR4 , I don't expect any new SR

**Comment by AlleSchonWeg** [comment]
For many issues there are existing PR's, but nobody reviewing these PR's. Perhaps MS should give repository control to some community members.

**Comment by developer9969** [comment]
Sadly I dont think there will be any new SR in fact SR5 does not exists.. Sadly people are logging issues but clearly there are not going to get fixed in xamarin forms at all . They are busy with MAUI. We are in a really bad place at the minute...

Let alone those PR's, What a nightmare at the moment.

**Comment by Huaba93** [comment]
it is so frustrating... how can we develop trust for MAUI? Companies with hundreds of employees rely on Xamarin.Forms, but this kind of support can ruin companies. I think for my company we need to get rid of Xamarin.Forms. The last 12 months have been a disaster.

**Comment by thisisthekap** [comment]
TL;DR: I am looking forward to MAUI, but please give some love to Xamarin.Forms, and give us a working SR 5 which solves the regressions introduced in SR 4.

Long story:

I appreciate the effort which is taken to make sure that MAUI is going to be the awesome product we all are looking for.

But please: Give some love to Xamarin.Forms! This framework which is considered to be production ready is suffering from severe issues with a high number of PRs which do not get any attention. Furthermore, the current service release consists of regressions, which makes it unusable to many companies out there.

I understand that there won't be new features for Xamarin.Forms, and that issues with working workarounds do not receive priority.

But please give us a SR5 which consists of many fixes which are already lying around in the code base, and which fixes the regressions introduced in SR4.

**Comment by Tommigun1980** [comment]
@davidortinau You did reiterate that the support commitment for Xamarin.Forms holds, so please communicate with us regarding what is going on?

I think that most of us are not hobbyists but actual employees and owners of actual companies. We have to be able to trust the support commitments. And to us any future releases and whatever else that’s possibly fun to work on doesn’t really matter, we need the tools to function properly and being able to trust the support commitment. Support commitment isn’t just a phrase on paper but an actual **commitment** to ensuring that the technology works. You also have to understand that you are losing customers at a rapid rate here. We developers really don’t care or have much insight into your internal deadlines at Microsoft and whatever new features you have promised marketing to whatever new release — we should be shielded from that, and you should have a roadmap that ensures that we can work normally while you work on whatever you do in the background.

Thank you.

**Comment by davidortinau** [comment]
> But please give us a SR5 which consists of many fixes which are already lying around in the code base, and which fixes the regressions introduced in SR4.

Please link me to the issues that are blocking you, and PRs you need. We are behind on triage and working to catch up. At present queries don't show my regressions with SR4.

> We developers really don’t care or have much insight into your internal deadlines at Microsoft and whatever new features you have promised marketing to whatever new release

We don't have such deadlines or promises. Our features are driven by your feedback and research, weighed by the typical constraints we all work with. Our roadmap is public. Our marketing event dates are public as soon as plans are solid. Our shipping schedule is predictable, though we don't post dates for every single release.

We are making every effort to be transparent so you can make solid plans for your projects and businesses. We will continue to service Xamarin.Forms as we have committed.

**Comment by MagicAndre1981** [comment]
> Please link me to the issues that are blocking you, and PRs you need. We are behind on triage and working to catch up. At present queries don't show me regressions with SR4.

[this](https://github.com/xamarin/Xamarin.Forms/issues/14428) is a regression, but looks like UWP is ignored at all if future is WinUI

**Comment by Huaba93** [comment]
[This](https://github.com/xamarin/Xamarin.Forms/issues/14426
) is also regression and blocking for us

**Comment by Tommigun1980** [comment]
> > But please give us a SR5 which consists of many fixes which are already lying around in the code base, and which fixes the regressions introduced in SR4.
> 
> Please link me to the issues that are blocking you, and PRs you need. We are behind on triage and working to catch up. At present queries don't show me regressions with SR4.
> 
> > We developers really don’t care or have much insight into your internal deadlines at Microsoft and whatever new features you have promised marketing to whatever new release
> 
> We don't have such deadlines or promises. Our features are driven by your feedback and research, weighed by the typical constraints we all work with. Our roadmap is public. Our marketing event dates are public as soon as plans are solid. Our shipping schedule is predictable, though we don't post dates for every single release.
> 
> We are making every effort to be transparent so you can make solid plans for your projects and businesses. We will continue to service Xamarin.Forms as we have committed.

You were asking for regressions.
I have mailed the team several times pleading for a fix to the https://github.com/xamarin/Xamarin.Forms/issues/14429 regression which completely blocks demonstrating my app. SR4 broke NuGet packages for a lot of people (https://github.com/xamarin/Xamarin.Forms/issues/14417 etc. etc.) but there's a workaround for that.

I also sent a list of must-have fixes months ago when you were asking for must-have fixes, and you are a CC on those.
I'll reiterate again: The CollectionView is not usable right now as it has problems with scrolling, selecting items, virtualisation and auto scrolling. I linked to the reports for these, and I have provided repro projects for every issue. There's no correspondence in the bug reports or any movement after mailing about them (as you asked me to do). In addition to that I provided a long list of must-have fixes mostly to the SwipeView, most of which already have fixes in the form of pull requests but have not been approved yet. They have just been sitting there for **months**.

This process here really isn't working. You said MAUI is based on customer feedback and that you don't have really external pressure from marketing etc, so who exactly asked for such a clean break that the current version is essentially abandoned? The thing I see people complaining about, to the degree of dropping Xamarin.Forms, is its bugs, not lack of features.
If despite this, what people really complain about, your roadmap is based on a clean break and not fixing issues for a long time I honestly don't think you have a good pulse on what is going on in the community.

Thank you.

**Comment by Tommigun1980** [comment]
And I concur with the posts in this thread saying that Microsoft has to start using Xamarin internally. Nobody can develop technology in a vacuum which is clearly happening here. Issues are not found and the pain is not evident in test apps with just one button. The question also arises that if Microsoft teams internally don't trust Xamarin, how can you convince developers to jump on board? Bugs happen when a lot of things are going on and are often not reproducible in a very monolithic repro project, so actually using your own tools would go a long way. But then again, I have made tens of repro projects for issues that are completely breaking my app and they have been sitting there for months (especially with the CollectionView, please refer to the email I sent a long time ago when you were soliciting for must-have fixes) without any action on them, so I don't know if it would help in the end. I don't really know what would help at this stage other than taking half a year off new development and just fixing bugs. There has to be some will behind it even if it may not be that exciting to work on.

**Comment by developer9969** [comment]
@Tommigun1980  Thank you for producing and submitting so many repos that highlights all the issues we have encountered.
@davidortinau  we have done all the workarounds in the world to make sure that we can have something that we can deploy but the collectionview took it to another level..  
Below is just a sample of the issues we face with collection view, 

- Does not resize properly
- Scrolling issues
- Empty view not working as it should
- Grouping issues

All things that you are aware of, even one of your team lead referred as the collection view as Frustrating in one of the chats...

SR4 
was not at all the kind of release we were expecting as it didnt contain any of the relevant fixes.

You know the collection view has issues can you prioritize and start cleaning the issues in the that section?

https://github.com/xamarin/Xamarin.Forms/issues/14429
https://github.com/xamarin/Xamarin.Forms/issues/14239
https://github.com/xamarin/Xamarin.Forms/issues/14224
https://github.com/xamarin/Xamarin.Forms/issues/14213

There is a lot of frustration here as many of us put the neck on the line to go for Xamarin instead of flutter etc.. and MAUI will be great but in the meantime we are stuck with a below par product.  Sorry for the harsh words but its a reality

Microsoft will never use Xamarin for their own product -- hopefully MAUI will change things....

Can you give us a SR5 with fixes that matters?

Many thanks

**Comment by Tommigun1980** [comment]
> @Tommigun1980 Thank you for producing and submitting so many repos that highlights all the issues we have encountered.
> @davidortinau we have done all the workarounds in the world to make sure that we can have something that we can deploy but the collectionview took it to another level..
> Below is just a sample of the issues we face with collection view,
> 
> * Does not resize properly
> * Scrolling issues
> * Empty view not working as it should
> * Grouping issues
> 
> All things that you are aware of, even one of your team lead referred as the collection view as Frustrating in one of the chats...
> 
> SR4
> was not at all the kind of release we were expecting as it didnt contain any of the relevant fixes.
> 
> You know the collection view has issues can you prioritize and start cleaning the issues in the that section?
> 
> #14429
> #14239
> #14224
> #14213
> 
> There is a lot of frustration here as many of us put the neck on the line to go for Xamarin instead of flutter etc.. and MAUI will be great but in the meantime we are stuck with a below par product. Sorry for the harsh words but its a reality
> 
> Microsoft will never use Xamarin for their own product -- hopefully MAUI will change things....
> 
> Can you give us a SR5 with fixes that matters?
> 
> Many thanks

Thank you. If we go down to detail level for posterity here's my must-have fixes:

CollectionView:
https://github.com/xamarin/Xamarin.Forms/issues/14429 - Regression where CollectionView starts randomly jittering around (this makes it break the entire app)
https://github.com/xamarin/Xamarin.Forms/issues/13592 - CollectionView initial selection is broken (ie you can't highlight items in a CollectionView)
https://github.com/xamarin/Xamarin.Forms/issues/13547 - CollectionView can't be scrolled to bottom when data is loaded (so can't really make a chat application etc. with it)

SwipeView:
https://github.com/xamarin/Xamarin.Forms/issues/12518 - SwipeView is so sensitive it opens while scrolling (makes it hard to scroll the CollectionView as SwipeViews constantly open)

SwipeView fixes with pull requests - these have pull requests that haven't been merged for ages:
**(Shouldn't this be considered low-hanging fruit as the fixes are already done? Why not merge them in??)**
https://github.com/xamarin/Xamarin.Forms/issues/13496
https://github.com/xamarin/Xamarin.Forms/issues/13495
https://github.com/xamarin/Xamarin.Forms/issues/13232
https://github.com/xamarin/Xamarin.Forms/issues/13491
https://github.com/xamarin/Xamarin.Forms/issues/12614
https://github.com/xamarin/Xamarin.Forms/issues/11795
https://github.com/xamarin/Xamarin.Forms/issues/11720

Nice-to-have fixes:
https://github.com/xamarin/Xamarin.Forms/issues/12590 - LineBreakMode="TailTruncation" doesn't work with Label.FormattedText
https://github.com/xamarin/Xamarin.Forms/issues/12704 - Editor with AutoSize="TextChanges" - text overflows control without it scrolling, forcing one to type blind

This is just scratching the surface, but just fixing those and merging the pull requests would go a long way in making my app work properly. This is **core** functionality mind you, like being able to scroll and select items in a CollectionView! It is almost impossible to make apps with these kinds of bugs, so it scares me that all focus is on the shiny new thing with these kinds of issues open (not to mention the 3.2K other open issues).

I also apologise for being so blunt about this, but some change has to happen soon and we are reliant on Xamarin.Forms working.

Thank you again!

**Comment by Tommigun1980** [comment]
Thank you. If we go down to detail level for posterity here's my must-have fixes:

CollectionView:
https://github.com/xamarin/Xamarin.Forms/issues/14429 - Regression where CollectionView starts randomly jittering around (this makes it break the entire app)
https://github.com/xamarin/Xamarin.Forms/issues/13592 - CollectionView initial selection is broken (ie you can't highlight items in a CollectionView)
https://github.com/xamarin/Xamarin.Forms/issues/13547 - CollectionView can't be scrolled to bottom when data is loaded (so can't really make a chat application etc. with it)

SwipeView:
https://github.com/xamarin/Xamarin.Forms/issues/12518 - SwipeView is so sensitive it opens while scrolling (makes it hard to scroll the CollectionView as SwipeViews constantly open)

SwipeView fixes with pull requests - these have pull requests that haven't been merged for ages:
**(Shouldn't this be considered low-hanging fruit as the fixes are already done? Why not merge them in??)**
https://github.com/xamarin/Xamarin.Forms/issues/13496
https://github.com/xamarin/Xamarin.Forms/issues/13495
https://github.com/xamarin/Xamarin.Forms/issues/13232
https://github.com/xamarin/Xamarin.Forms/issues/13491
https://github.com/xamarin/Xamarin.Forms/issues/12614
https://github.com/xamarin/Xamarin.Forms/issues/11795
https://github.com/xamarin/Xamarin.Forms/issues/11720

Nice-to-have fixes:
https://github.com/xamarin/Xamarin.Forms/issues/12590 - LineBreakMode="TailTruncation" doesn't work with Label.FormattedText
https://github.com/xamarin/Xamarin.Forms/issues/12704 - Editor with AutoSize="TextChanges" - text overflows control without it scrolling, forcing one to type blind

This is just scratching the surface, but just fixing those and merging the pull requests would go a long way in making my app work properly. This is **core** functionality mind you, like being able to scroll and select items in a CollectionView! It is almost impossible to make apps with these kinds of bugs, so it scares me that all focus is on the shiny new thing with these kinds of issues open (not to mention the 3.2K other open issues).

I also apologise for being so blunt about this, but some change has to happen soon and we are reliant on Xamarin.Forms working.

Thank you again!

**Comment by Tommigun1980** [comment]
I am not kidding when I say that nowadays 80% of my time goes to workarounds for Xamarin.Forms bugs and 20% goes to actual development. My app is fairly mature so it is all about workarounds now. This is untenable.

**Comment by AndyDentFree** [comment]
@davidortinau wrote back in April

> .NET MAUI is Xamarin.Forms vNext in this context. What time frame for transitioning would be comfortable for you? What would you expect to happen in Xamarin.Forms when iOS 15, 16, 17 ships?

Seriously?
1. A **five** year commitment to maintenance
2. A team status page so we can see how many engineers you have assigned
3. A core set of good samples showing a range of dev styles, not just Shell, that is actively maintained so it always works with new releases of Forms
4. Preferred - blog posts about significant internal tools used in MS either in Forms or Maui so we know people who decide how big engineering teams should be are dogfooding this. If they dont' feel pain, it won't be a priority for them. I don't think it reasonable to expect these all to start in Forms but it would be **very reassuring** to hear of at least one tool in regular management use that is in Forms and then to hear of its migration to Maui.

I'm a developer with 40 years experience, much of which is the MS ecosystem. So, yeah, I've seen every single generation of _next best thing_ come and go. 

I was a developer for nearly two years on the Realm C# SDK (now part of MongoDB) so I saw Forms support from the library vendor perspective. I've worked in enterprises shipping to tens of thousands of end users on Windows. I have a good idea of technology migration horizons.

I have a legacy project in forms which I'm diving back into and a new family side-project. In my main startup, I have to pick a tech within months for a cross-platform app.

Right now I'm very seriously contemplating using Flutter for the family project, so I can decide if I will recommend it unilaterally as a replacement for Forms/Maui for mobile development.

Note my emphasis on **reassuring** in this response. In some degree of shock from this thread, I'm not feeling anything like that. I deeply respect much of the work done at MS and that it's not decisions made at your level that affect the resources.

The responses in this thread should show that people have lost trust and that loss of trust extends to not trusting much in the Maui migration either.

**Comment by DeerSteak** [comment]
At a minimum getting these fixes as mentioned by @Tommigun1980 and supporting the current shipping versions of the AndroidX libraries should happen. As it stands, it appears Xamarin.Forms 5 is abandonware and there's no replacement. It's extremely frustrating to have the latest fixes and then get compiler warnings like these:

Warning	NU1608	Detected package version outside of dependency constraint: Xamarin.Forms 5.0.0.2083 requires Xamarin.AndroidX.Lifecycle.LiveData (>= 2.2.0.4 && < 2.3.0) but version Xamarin.AndroidX.Lifecycle.LiveData 2.3.1.1 was resolved.	

Warning	NU1608	Detected package version outside of dependency constraint: Xamarin.Forms 5.0.0.2083 requires Xamarin.AndroidX.RecyclerView (>= 1.1.0.8 && < 1.2.0) but version Xamarin.AndroidX.RecyclerView 1.2.1.1 was resolved.	

Warning	NU1608	Detected package version outside of dependency constraint: Xamarin.Forms 5.0.0.2083 requires Xamarin.AndroidX.Core (>= 1.3.2.3 && < 1.4.0) but version Xamarin.AndroidX.Core 1.6.0.1 was resolved.	

And then on top of it to have bugs like Shell on Android disposing custom renderers with no workaround is extremely frustrating. 

https://github.com/xamarin/Xamarin.Forms/issues/9073

That's been around for almost two years. You make all these new technologies and beg us to use them, and then don't support them. As @AndyDentFree says, that kills trust in the Maui migration.

**Comment by AlleSchonWeg** [comment]
@davidortinau
As a good starting point you should merge the existing PR with many fixes.

**Comment by wowpravin** [comment]
@davidortinau imho asking your customers 'now' what PRs are critical is not a good state of affairs. Honestly if Microsoft was using XF internally then you would already know what to fix!

**Comment by Tommigun1980** [comment]
I also think that you need integration tests; maybe some kind of complex test app that touches upon most areas used in an actual app, which you also always run manually before committing changes as a smoke check. Just throw in all kinds of weird stuff in there. Having a billion unit tests is nice but they are not reflective of real world usage at all. A button almost always works **in isolation**, but combine that with other components and things start breaking down. There is no way you could create so many unit tests that they could catch all possible real-world usage scenarios, without any manual testing/smoke checking and integration testing. Also, when new features are released the developers have to test them in the toughest possible setting, on ALL supported platforms. TRY to break it! Don't settle for a test app which houses only that one new fancy button. So yeah, very often the components work pretty well in isolation but not when used in an actual app.

But then there's components like the CollectionView where even core concepts like selection and 'scroll to bottom' doesn't work. I think that you have to start catching those kinds of issues before releasing them to start gaining trust back. I honestly have a hard time figuring how for example the CollectionView was tested... When features were added to it (like selection capability and it firing events for various actions), was the code that was written manually tested? I don't mean artificial tests, but some kind of test app that houses the component in question, where every new feature is tested while being developed? Because so much of its core features are broken in the simplest possible repro case, that I honestly would like to know if those features were tested?

Once again, apologies for the bluntness, but we need major changes to the process in order to continue investing in Xamarin.Forms/MAUI.
Thank you.

**Comment by timahrentlov** [comment]
Why aren't MS dogfooding like Google is with Flutter? 

And think of all the "We want your feedback and input" avoidance speak we could get rid of :)

Really. Start using your own product.

**Comment by Huaba93** [comment]
It has often been said that MAUI becomes Xamarin.Forms Next, so the wheel is not being reinvented. Some things are made new (e.g. renderer-->handler) but some things are simply taken over. Or have I understood this wrong?

This also means that many problems known in Xamarin.Forms we will find in MAUI as well!

@davidortinau  What has changed that it should become so much better with MAUI?

**Comment by nacompllo** [comment]
This problem is very blocking for our application, it prevents updating the origin of the lists in macOS at runtime, it is very frustrating that something so basic is failing which forces us to look for laborious temporary patches with the loss of time that it causes. 
[N.º 14155](https://github.com/xamarin/Xamarin.Forms/issues/14155)

**Comment by jsuarezruiz** [comment]
As Gerald mentioned, we are here, we read your feedback and we want to have regular releases improving. It is impossible to attend to everything at once, I hope it was possible!, but we will try to create boards with the issues, status, explain the priorities established etc. We will also update with progress status.

We keep moving forward, but I want to share 3 PRs of 3 issues mentioned here: https://github.com/xamarin/Xamarin.Forms/pull/14565, https://github.com/xamarin/Xamarin.Forms/pull/14567 and https://github.com/xamarin/Xamarin.Forms/pull/14565 There will be a NuGet package associated with each PR so if you can test and give feedback it would be great!

Again, thank you all for your feedback and help to improve Xamarin.Forms!.

**Comment by jfversluis** [comment]
Just to reiterate and add to what Javier is saying: we'd love to fix all the issues, but we can't. We'll do our best to get out as many as possible. I'm working on a more structured plan with regular, predictable releases and I will be the one dedicated to only pushing those out and keeping you updated. As soon as I can share more concrete things about this I will.

Having that said; keep adding things to this issue, I will definitely keep an eye on it, at least until we work through the initial list of issues that we gathered here. Just know that it's not a guarantee that they will be picked up with priority or will be fixed at all.

Also, I'd really appreciate it if you could go over the issues that are important to you and try out the NuGets on the PRs, that will help speed up getting them merged for sure!

Thanks all!

**Comment by MagicAndre1981** [comment]
It would be also nice to add #13150 to next SR 🤞

**Comment by developer9969** [comment]
@jfversluis @jsuarezruiz  Having some feedback is a massive improvements. Thank you!!
It's understandable that you cannot fix everything as there have been years of neglect, but I would like to point out  that one of the major component "CollectionView" is causing lots of problem in delivering quality apps .
This  is in my view is one of the most used and important component after the grid, we dont expect to be perfect but usable, "Scrolling-grouping-selection-footers-resizing -emptyview etc.." those are basic functionalities that make or break an app!!.

Everyone will have a set of bugs that is important to them but I guess that if a priority has to be set than the CollectionView should be at the very top, and if not fixed it will be just the same with MAUI.

I hope this release will include fixes to the collectionview.

Thank you again for the feedback

**Comment by jfversluis** [comment]
Thanks @developer9969 I wouldn't call it neglect, but let's not go into discussions about what has happened in the past. We can't change that :)

I hear you about the `CollectionView` and that definitely has our attention. We would love for you to use it instead of the `ListView` and everything we fix for the `CollectionView` today will also benefit .NET MAUI. I think a lot of the issues in this issue (issue-ception...) are also about the `CollectionView` that will be a area of focus, but thanks for underlining the importance!

**Comment by jsuarezruiz** [comment]
More PRs from issues mentioned in this thread: https://github.com/xamarin/Xamarin.Forms/pull/14572, https://github.com/xamarin/Xamarin.Forms/pull/14573

**Comment by jsuarezruiz** [comment]
The message from this issue https://github.com/xamarin/Xamarin.Forms/issues/14409 is similar, and I want to associate them and share the same message everywhere: we are here, we want to improve as much as possible and we will openly inform everything.

**Comment by AndreasReitberger** [comment]
@jsuarezruiz @jfversluis 
Thank you for getting this repo back to life :)

Maybe you also can look into this issue. It came up with SR4, as far as I can remember.
https://github.com/xamarin/Xamarin.Forms/issues/14510

Thank you,
Andreas

**Comment by MagicAndre1981** [comment]
RP #14506 should be also added as this will fix compile issues with latest Xamarin google material nugets.

And I've seen a bot here on github that does nuget updates, I see that XF has a lot of pending nuget updates so maybe adding the bot here would help detecting such major issues like with Xamarin google material nuget

**Comment by jfversluis** [comment]
> RP #14506 should be also added as this will fix compile issues with latest Xamarin google material nugets.
> 
> And I've seen a bot here on github that does nuget updates, I see that XF has a lot of pending nuget updates so maybe adding the bot here would help detecting such major issues like with Xamarin google material nuget

I was going to check on that one and make sure to see if it needs to go in. Thanks for the bot suggestion, we're aware but I don't think it's something we would want to do automated :)

> @jsuarezruiz @jfversluis
> Thank you for getting this repo back to life :)
> 
> Maybe you also can look into this issue. It came up with SR4, as far as I can remember.
> #14510
> And those:
> #13510
> 
> #12141
> 
> Thank you,
> Andreas

The first one is already fixed and should be in the next release, I've added the other two to the [next service release version board](https://github.com/xamarin/Xamarin.Forms/projects/88)

**Comment by jfversluis** [comment]
These issues were mentioned in #14409

#11993, #5265, #14363

Adding them here and closing the afore mentioned issue so we can keep things a bit centralized

**Comment by YiannisBourkelis** [comment]
Nice to see that MS devs are focusing again on solving Xamarin.Forms bugs. We need this to keep working on this framework. I have also run into several issues with keyboard handling, but I was able to find workarounds expect this one I had posted here, about keyboard dismiss on key press: https://github.com/xamarin/Xamarin.Forms/issues/12945

Keyboard issues are also reported every now and then from other developers here.

**Comment by melimion** [comment]
It would be great to fix dynamic sizing issues in CollectionView:

https://github.com/xamarin/Xamarin.Forms/issues/9520
https://github.com/xamarin/Xamarin.Forms/issues/11011
https://github.com/xamarin/Xamarin.Forms/issues/13781

And some CollectionView initialization issues:
https://github.com/xamarin/Xamarin.Forms/issues/9079
https://github.com/xamarin/Xamarin.Forms/issues/13675

**Comment by MagicAndre1981** [comment]
> > RP #14506 should be also added as this will fix compile issues with latest Xamarin google material nugets.
> > And I've seen a bot here on github that does nuget updates, I see that XF has a lot of pending nuget updates so maybe adding the bot here would help detecting such major issues like with Xamarin google material nuget
> 
> I was going to check on that one and make sure to see if it needs to go in. Thanks for the bot suggestion, we're aware but I don't think it's something we would want to do automated :)

from what I see the bot generates PRs that have to be reviewed and merged, so you still have the control

**Comment by angelru** [comment]
I think this is also important:
[10085](https://github.com/xamarin/Xamarin.Forms/issues/10085)

**Comment by DDFLotz** [comment]
Great to see that SR5/6 is happening :) 
Down below is a list of important issues for our team

TitleView related: 
#12382 (links to a working PR)
CollectionView related:
#11299
#6554 (open since 2019)
TabbedPage related: 
#11301

it would be awesome if some of them could be addressed and updated, thanks!

**Comment by ondrejsv** [comment]
#6554: I second that.

**Comment by eramrit78** [comment]
#13229 this one also

**Comment by workgroupengineering** [comment]
I have a suggestion. You can create a section in the "Priority Issue" discussions and replicate or transfer issues that are considered important.
The advantage of this solution is that it is possible to vote the discussion in order to avoid duplicates and get the situation quickly.

**Comment by jfversluis** [comment]
Thanks for the suggestion @workgroupengineering I'm not quite sure if I understand what you mean though :)

**Comment by jfversluis** [comment]
Hey everyone! We're about to release a new service release version. From here on out our aim will be to release one every month until the end of life of Xamarin.Forms :) I will keep creating new projects and set a target date on it, it might not be perfect each month, but should be around that date. First focus is still low-hanging fruits and issues mentioned here. They already seem concentrated on `CollectionView` but `CollectionView` will definitely be a big focus for us as well.

A lot of you have already been cooperating by testing out the fixes in PRs and such. A big thanks for all of you, let's keep up the great work together! <3

**Comment by DeerSteak** [comment]
Glad it's not going to be another five month wait like it was between SR3 and SR4. We get that MAUI is the future, but we live in the present and have to ship apps in the present.

**Comment by AndreasReitberger** [comment]
@jfversluis @jsuarezruiz 
Thank you for your great effort the last weeks. Looking forward to the next releases.

I'm just curious, is there a reason while the updates get released while not all bugs are fixed linked in the projects?

SR4: https://github.com/xamarin/Xamarin.Forms/projects/86
SR5: https://github.com/xamarin/Xamarin.Forms/projects/87

Are this bugs still on your list?

Thank you 🙏

**Comment by tschramme86** [comment]
Thank you for your continuous effort here! Really appreciate

Since I just saw that there are several Shell Navigation Bugs on the list to fix in SR6, maybe you can also check if this one is possible in SR6? https://github.com/xamarin/Xamarin.Forms/issues/12958

**Comment by jsuarezruiz** [comment]
@AndreasReitberger They continue to be taken into account in upcoming releases.

**Comment by jfversluis** [comment]
@DeerSteak SR5 should have all the Google dependencies updated to the latest bits at this time.

@AndreasReitberger I'm a bit behind on the administration but the idea is to move unfinished issues to newer boards. However, be aware that we still are in charge of the priorities which might not align with yours.

Thanks everyone for the positive notes!

**Comment by Crotalus** [comment]
Here's another issue that cause problems for people still using ListView:

Regression with ListView using with CachingStrategy RecycleElement causes massive lag in debug mode with hot reload:
(5.0.0.2012 broke it, works in XF 5.0.0.1905)

Issues: #14069, #14218, #14190, #11344

**Comment by jfversluis** [comment]
@davidaramantSEP yes it is :) the support will be 12 months from the .NET MAUI GA date

**Comment by tranb3r** [comment]
When is SR5 (5.0.0.2117) nuget going to be released ?

**Comment by AndyDentFree** [comment]
> the support will be 12 months from the .NET MAUI GA date

That's good but not confidence-inspiring news.

Bare minimum which would restore my confidence would be **12 months from .Net MAUI _STABLE_ state** with a clear and detailed explanation agreed on by the community on what we consider _stable_ including a commitment to which demo applications MS will commit to keeping working as a _smoke test_ of MAUI.

Much less than that leaves us floating in a sea of doubt as to what's a safe recommendation for new projects.

**Comment by davidaramantSEP** [comment]
I agree that I think Microsoft needs to do something extra to inspire some more confidence in the platform. The experience with Xamarin for the last year has really called into question if we should be recommending Xamarin to clients for new projects. At a minimum, we need to have a hard discussion with the client about what we've seen. Support going AWOL for months on end is simply unacceptable.

You are _saying_ the right things, which is great, but that only goes so far. Being more transparent on what efforts Microsoft is taking to improve the quality of Xamarin & Maui would be very appreciated. Relying entirely on the community to report bugs isn't good enough - some of the issues with `CollectionView` (for example) were so dire it should have never have left the "experimental" state and certainly shouldn't have been the recommended go-to solution in the documentation.

**Comment by jfversluis** [comment]
> When is SR5 (5.0.0.2117) nuget going to be released ?

It should have been released already but during our quality check we found a regression that we wanted to fix. I'm creating a new build as I am typing this release should be ASAP. Following releases should be more smooth.

@AndyDentFree Sorry you feel that way, nothing much I can do about it though. 

@davidaramantSEP I'm doing everything I can to bring back at least some of that confidence. I am being transparent and very frequently updating you. And you're right: I'm saying the right things, now I have to do them, so instead of having discussions here that's what I will redirect my efforts to. I hope that will be enough.

**Comment by AndyDentFree** [comment]
> I'd love to see demo apps that serve as a smoke test.

This is a topic dear to my heart. I worked for nearly two years at Realm, on the C# SDK for their mobile database. So, I've been where you are now (without the amount of technical debt). Part of my job included tech support and also doing releases.

We had some trivial smoke test apps and _building them from scratch_ with each release was our way to validate that the latest Xamarin Forms toolchain hadn't broken something.

By _from scratch_ I mean we had the source code ready to copy-and-paste but created a brand new app each time. Painful experience proved that just recompiling an app with the latest build is **not** the same as validating people's experience of building new apps.

This may sound onerous but, when things work, it's about 10 minutes per platform, per release.

**Comment by jfversluis** [comment]
Thanks for sharing from your experiences!

**Comment by varyamereon** [comment]
My ten cents, #14505 is a blocking issue for me as I've been unable to find a workaround and it means the app basically doesn't work on iOS 15. Would think this affects a large number of people. Same as https://github.com/dotnet/maui/issues/1654.

**Comment by tranb3r** [comment]
This one also deserves to be fixed : #8383

**Comment by Amenti** [comment]
We get a lot of bad customer feedback due to the broken scrolling of Editors beginning with iOS 14, see #13040, #2233, #4695, #12695. :(

**Comment by dan5602** [comment]
> My ten cents, #14505 is a blocking issue for me as I've been unable to find a workaround and it means the app basically doesn't work on iOS 15. Would think this affects a large number of people. Same as [dotnet/maui#1654](https://github.com/dotnet/maui/issues/1654).

This is also a huge issue for us. iOS 15 is unusable. Our app uses Shell. Customers who have upgraded are already complaining - This issue surely needs some priority? How was this not found during all of the betas?

**Comment by jfversluis** [comment]
Just a quick update here: we have released our first service release yay! 🤗 

From here we are going to release one every month until end-of-life, no matter how big or small the release will be. I will keep creating project boards and keep moving the list along and hopefully it should get shorter as more fixes go in. 

We still could use your help! You can do so by testing out the fixes from PRs as we publish those. Also, if you have a solution in a custom renderer right now, or you have pinpointed a cause of a bug, be sure to let us know, that will help get your bug fixed as well.

Priority at the moment will be `CollectionView`, Shell and any quick fixes we can do easily or might already be in a PR.

Thanks for your patience and on to the next release!

**Comment by dan5602** [comment]
Below is a custom render that fixes the issue I'm having with the nav bar background colour. Not perfect but will allow me to get a release out. If anyone has a better workaround for now...please share!

`using UIKit;
using Xamarin.Forms;
using Xamarin.Forms.Platform.iOS;

[assembly: ExportRenderer(typeof(APPNAME.AppShell), typeof(APPNAME.iOS.Renderers.CustomShellRenderer))]
namespace APPNAME.iOS.Renderers
{

    public class CustomShellRenderer : ShellRenderer
    {
        protected override IShellSectionRenderer CreateShellSectionRenderer(ShellSection shellSection)
        {
            var renderer = base.CreateShellSectionRenderer(shellSection);
            
            if (renderer != null)
            {
                if (renderer is ShellSectionRenderer shellRenderer)
                {
                    
    
                    var appearance = new UINavigationBarAppearance();
                    appearance.ConfigureWithOpaqueBackground();
                    appearance.BackgroundColor = new UIColor(red: 0.86f, green: 0.24f, blue: 0.00f, alpha: 1.00f);
                    
                    appearance.TitleTextAttributes = new UIStringAttributes() { ForegroundColor = UIColor.White};
                    
              
                    shellRenderer.NavigationBar.Translucent = false;
                    shellRenderer.NavigationBar.StandardAppearance = appearance;
                    shellRenderer.NavigationBar.ScrollEdgeAppearance = shellRenderer.NavigationBar.StandardAppearance;
                    
                }
            }

            return renderer;
        }

        protected override IShellItemRenderer CreateShellItemRenderer(ShellItem item)
        {
            var renderer = base.CreateShellItemRenderer(item);
            
            if (renderer != null)
            {
                if (renderer is ShellItemRenderer shellItemRenderer)
                {
                    var appearance = new UITabBarAppearance();
                    
                    appearance.ConfigureWithOpaqueBackground();

                    shellItemRenderer.TabBar.Translucent = false;
                    shellItemRenderer.TabBar.StandardAppearance = appearance;
               
                }
                

            }

            return renderer;
        }
        
    }
}`

**Comment by AlleSchonWeg** [comment]
Hi @jfversluis ,
i have a question. ;)
What happens with the open issues in the XF github repository, if XF support comes to an end? Will the repository deleted and the issues transfered to maui? Or all issues will be fixed before the repository is closed? 😎
At the moment i have 10 issues open and around 50 issues/pr which i'm watching and hoping for a fix.

Thx

**Comment by jfversluis** [comment]
Good question! Thank you for that.

I think it's clear we're not just going to blindly move over what we have here to the .NET MAUI repo. Then we will already be under water even before we started, that's not going to be fun for anyone. And although I would love to fix them all before we leave Forms behind, I think that is not realistic as well.

Together with pushing out Service Releases I'm also trying to go through the list and see what is still relevant. That is a hard thing to do because it's impossible to test everything that is in there. There is a good chance that there are issues that are outdated. Either because too much time has passed or the bug is fixed, either in .NET MAUI or Xamarin.Forms. For the issues that we are fixing in Forms during the remaining lifetime will go on a list so that we can verify against .NET MAUI and we don't have regressions there.

However, there definitely will be bugs that are still relevant on the Xamarin.Forms repo. To be very honest I don't have an actionable plan for it right now. If you are willing and able to help out and verify Forms vs .NET MAUI I can only ask to test things on .NET MAUI and let me know the results on the issues here so I can give them a label or, even better, close them if they are already fixed.

For all the rest; if you have any (realistic, achievable) suggestions to handle this, I am definitely open to ideas!

Additionally,  thinking about this, let me break up the issues: bugs and features.

Bugs should be fixed, but see al of the above. Features are definitely not going to be added anymore. So as a rule of thumb: any issue or PR that will add new functionality or APIs to Xamarin.Forms are definitely not going to be added anymore. If you have those on your list and they are important to you I would like to suggest you open a new issue on the .NET MAUI repo with all the (updated) details on what you would like to add and add a link to the Forms issue.

I am curious about your list though. Could you post it here or send it to my email? To be clear: me asking means absolutely nothing in terms of things going to be solved. Just curious what the things are that are important to you. Thanks!

Does this help at all?

**Comment by AlleSchonWeg** [comment]
Hi,
email sent!
Thank you.

**Comment by AndyDentFree** [comment]
A modest economic proposal responding to @jfversluis's comment on bug fixes.

Quick version:

1. Build one or more _big_ sample apps that show the most important bugs, to also become smoke test for MAUI.
2. Use interns/cheap outsourced labor for initial review, building a simpler central list linking issues to 1.

Context:

1. Strategically, MAUI needs to be robust - this & other issue threads demonstrate community concerns
2. The legacy of Forms bugs & age of many reports means awkward verification - need a more central status tracker
3. A single or very small set of demo apps as smoke tests means it's easy for MAUI builds to be verified and also ongoing detection of Forms regressions. 
4. Expensive, relatively senior people shouldn't be getting bogged down in issue review.


**Related idea** - make it a new interview question for _anyone recruiting Xamarin Forms developers_ - they have to verify an issue. Shows ability to build, communicate & diagnose.


**Demo App size philosophies**

There are two reasons to have demo/test apps. One is to explore or prove a technology & these are best kept as small as possible so maintenance is lightweight. Someone needing to explore that tech in future wants the least possible effort in updating a sample, especially if they are trying out preview/alpha versions of frameworks.

Smoke test validators to build confidence that a complex, realistic scenario is both buildable and works need to be bigger.

Frameworks like Forms need both.

 (I'm trying to be more helpful than my previous disappointed rants.)

**Comment by jfversluis** [comment]
Thanks for your input!

**Comment by MitchBomcanhao** [comment]
@AndyDentFree an issue with interns reviewing issues is that they don't know the product at all, and so will create more churn for people trying to report the problems (and the interns not understanding the problem) and then the more experienced/competent people will still have to review whatever stuff the interns did/didn't do because they can't understand the problems enough to effectively review them, or miss out on important issues being excluded/ignored by the interns.

what you're describing makes me think of the microsoft answers forum where the microsoft "experts" have no knowledge of the products and just regurgitate the same answers over and over, even when they clearly do not apply to the problem being discussed.

what you're proposing will effectively just make things more difficult as we will have to convince some middleman who has no idea about the technical stuff that our problem is indeed a problem and needs to be looked at by someone higher up.

**Comment by tranb3r** [comment]
It does not matter how many open issues there are. The important thing is to make sure it's easy to identify high priority issues. But I'm not sure it's the case right now.

A few numbers:
- open issues : 3249
- bugs : 2343
- bugs with high impact : 114
- bugs with high impact and more than 5 reactions : 53
- bugs with high impact and more than 10 reactions : 24

Maybe the most impactful thing to do would be to review (and fix) those 24 'high priority' issues.
[label:"t/bug :bug:" is:issue is:open label:"m/high impact :black_large_square:" reactions:>10](https://github.com/xamarin/Xamarin.Forms/issues?q=is%3Aissue+is%3Aopen+label%3A%22m%2Fhigh+impact+%3Ablack_large_square%3A%22++label%3A%22t%2Fbug+%3Abug%3A%22++reactions%3A%3E10+)

What do you think ?

**Comment by jfversluis** [comment]
Thanks for the input @tranb3r. Have you also actually checked the issues that are attached to this label? 

By opening your link I already see a couple that will definitely not be making it into any upcoming Service Release and depending on some factors these might not be high impact (anymore). So that would mean still go trough all the issues and reevaluate their status.

Conversely, there are also issues that are definitely not marked high impact, like a couple of issues that make Forms apps not-usable with iOS 15, that would be on this list, but they still are on our current radar and fixed with high priority.

Additionally, there is also the labels marked with `i/high` (or low, etc.) so the labels might not even mean what you think. Also I think there is more to just consider than the impact, like the offset with how much effort is it to fix this issue?

Is this all a great situation to be in or any excuse? Absolutely not, we are to blame ourselves for this administration. But I think it's also not as simple as you're trying to state it here. Reading between the lines I see that you disagree with out approach? What are issues that are important to you that are not on our list right now?

**Comment by tranb3r** [comment]
@jfversluis I understand your point of view.

First, you're right, I'm probably wrong assuming that 'high impact' means 'high priority'. Maybe this could be clarified [here](https://github.com/xamarin/Xamarin.Forms/labels?sort=count-desc) by making sure all the most used labels have a description.

However, I'm sure you agree that it's unrealistic to think that one can review 3249 open issues.
So can you share with us what is the proper query to identify 'high priority' issues ? ('high priority' meaning the ones that need to be reviewed asap, and then either fixed or closed)
Also, why is SR4 project still open ? What about SR5 issues that have not been fixed ? Maybe you should clarify how theses projects are managed, because I do not understand what is exactly 'your list'.

**Comment by jfversluis** [comment]
We should have clarified under those labels what is meant. I think it's rather useless to do that retroactively. Since there was obviously no clear definition, different people might have used the label differently and thus the results will be no reflection of the truth.

The SR4 board is open because that one dates from before I (re)joined the team and I still wanted to evaluate what is on there and move things over to the [current board](https://github.com/xamarin/Xamarin.Forms/projects/88). As stated in this issue: I will create a project board per Service Release, and that is what is "on the list". Besides what was already deemed important in earlier releases, I have added all the issues that are listed in this issue and we have a special focus on Shell and CollectionView as we feel that is what a lot of people are using and running into blockers. In addition, we have a lot of "low hanging fruit" in the form of PRs that are already open, so we're evaluating what we can salvage from that and merge it.

How the projects are managed; I move everything that is not in the Done column to the next project, until, ideally, all columns are empty. The ones that stay in Done, we can see what was done for that SR, the other issues can be understood as: on our list/todo.

Feel free to highlight any issues that might be missing according to you and we'll see what we can do about them. Or, even better, in the spirit of Hacktoberfest, feel free to help out so there is a better chance of getting things fixed that are important to you!

**Comment by jfversluis** [comment]
Note: I'm not claiming to have exclusive right to the truth here. I wish it was as simple as filtering the list and just starting from the top, but unfortunately it's not, as hopefully I can make clear with what I'm writing here. I am trying to be as transparent and open as I can because I love to work with (all of) you to make your experience the best we can. I hope you won't conceive it as rejecting any idea that is not mine, that is absolutely not what I'm trying to do ☺️

With just the people assigned to this we're doing the best we can and everything we can (just have a look at all the PRs that Javier has put out in a short time!). Help is more than welcome in any shape or form. Either point out issues that are important to you, issues that are not relevant anymore so I can close them or anything else. I'm hoping to do this together and make our final ride into the Xamarin.Forms sunset the most pleasant one for everyone.

**Comment by tranb3r** [comment]
Here is a list of the issues I'd like to be fixed, that are not already in the board: #9839 #14154 #3054 #3855 #11593
Thank you @jfversluis and keep up the good work !!

**Comment by jfversluis** [comment]
Awesome, thank you @tranb3r! I'll go through them and see what we can do :)

**Comment by jfversluis** [comment]
Hey @somoreingold I can totally understand! I will have a look at those and see what we can do.

**Comment by jfversluis** [comment]
Hey everyone! Just to update this thread, another SR has been released with a good number of fixes. We've been making good progress with a good number of fixes that are going in and the monthly releases.

I realize there are still some issues I need to gather from the last few comments here and put them on a board, no worries, I got that on my radar. I'm not sure how much sense it makes to keep this thread going much longer, I hope we've established some form of trusts and you appreciate what we're trying to do here :) 

If there is any concern feel free to reach out either publicly for transparency or privately if you prefer that. For now I will leave this open, but I might close it in the near future. If there are any other questions or comments, let me know!

**Comment by zzyzy** [comment]
Would appreciate if the following could be fixed :) 
#13597

**Comment by yunusefendi52** [comment]
Also CollectionView related issue that I am having #14522

**Comment by YiannisBourkelis** [comment]
Regarding Collectionview implementation, there is one more very important feature that is missing: view (or scroll indicator) inset when the keyboard appears. Here is the related issue on the mater: [https://github.com/xamarin/Xamarin.Forms/issues/9691](https://github.com/xamarin/Xamarin.Forms/issues/9691)

**Comment by jfversluis** [comment]
Hey everyone! In case you missed it, we just released Service Release 7 a few days ago. I see the communication is slowing down here and as much as I enjoy being in direct contact with you, I will close this for now so we keep a good overview of things. I hope with all that we've been doing that we have restored a bit of faith on your part that at least things are moving. Probably, in some cases, not as fast as you had hoped, but we'll get there.

I will keep creating project boards for each release so you can see what we're looking at. Feel free to ping me in any issue that you want to bring to my attention and I'll try to get back to you asap. My email is also open and maybe you want to join the [Discord server](https://aka.ms/dotnet-discord) if you want to discuss something.

Again, thank you all so much for your patience and also the collaboration! You have all been very helpful pointing out important issues, verifying fixes and more. Let's keep in touch! <3
